lib/features/sessions/cubit/bookmark_session_cubit.dart

```diff
@@ -34,6 +35,20 @@ class BookmarkSessionCubit extends Cubit<BookmarkSessionState> {
         bookmarkStatus: bookmarkStatus == BookmarkStatus.bookmarked,
       );
 
+      if (bookmarkStatus == BookmarkStatus.bookmarked) {
+        final session = await _dBRepository.getSession(sessionId);
+        final endTime = session?.endDateTime;
+        final notifcationTime = endTime?.subtract(const Duration(minutes: 5));
+
```

Comment:
Pass in the entire `session` object into `createScheduledNotification(` and then do all this nifty things in the service/repository function to keep the cubit lean.